src/Controller/AgendaInspectionController.php

```diff
@@ -102,6 +107,45 @@ public function create(Agenda $agenda, AgendaCaseItem $agendaItem, CprHelper $cp
 
             $digitalPostHelper->createDigitalPost($document, $inspection->getTitle(), get_class($agendaItem), $agendaItem->getId(), [], $digitalPostRecipients);
 
+            // Attach document to case
+            $case = $agendaItem->getCaseEntity();
+            $caseDocumentRelation = new CaseDocumentRelation();
+            $caseDocumentRelation->setCase($case);
+            $caseDocumentRelation->setDocument($document);
+
+            $entityManager->persist($caseDocumentRelation);
+
+            $case->addCaseDocumentRelation($caseDocumentRelation);
+
+            // Create case event (sagshændelse)
+            $caseEvent = new CaseEvent();
+            $caseEvent->setCaseEntity($case);
+
+            $caseEvent->setCategory(CaseEvent::CATEGORY_OUTGOING);
+            foreach ($inspection->getRecipients() as $recipient) {
+                $caseEventPartyRelation = new CaseEventPartyRelation();
+                $caseEventPartyRelation->setParty($recipient);
+                $caseEventPartyRelation->setCaseEvent($caseEvent);
+                $caseEventPartyRelation->setType(CaseEventPartyRelation::TYPE_RECIPIENT);
+                $entityManager->persist($caseEventPartyRelation);
+            }
+
+            $caseEvent->setSubject(CaseEvent::SUBJECT_HEARING_CONTRADICTIONS_BRIEFING);
+            $caseEvent->setReceivedAt(new DateTime('now'));
+            $caseEvent->setCreatedBy($this->getUser());
+
+            $digitalPost = $digitalPostRepository->findByDocumentAndAgendaCaseItem($inspection->getDocument(), $agendaItem);
+
+            if (1 !== count($digitalPost)) {
+                throw new CaseEventException(sprintf('Should find one DigitalPost but found %d', count($digitalPost)));
+            }
+
+            $digitalPost = reset($digitalPost);
+            $caseEvent->setDigitalPost($digitalPost);
```

Comment:
Why don't we just use the digital post created in line 108?

Reply:
`DigitalPostHelper::createDigitalPost` does not return the created digital post, but i suppose i could modify it to do so.

Reply:
It would make sense for it to return the created digital post. And help us not making assumptions on how to find it afterwards.

Reply:
Modified to return created digital post.